server/punishments.js

```diff
@@ -1526,8 +1527,9 @@ const Punishments = new (class {
 	checkNewNameInRoom(user, userid, roomid) {
 		/** @type {Punishment?} */
 		let punishment = Punishments.roomUserids.nestedGet(roomid, userid) || null;
-		if (!punishment && Rooms(roomid).parent) {
-			punishment = Punishments.checkNewNameInRoom(user, userid, Rooms(roomid).parent.id);
+		const room = Rooms(roomid);
```

Comment:
```suggestion
		const room = !punishment && Rooms(roomid);
```

Save a call if the punishment doesn't exist.

Reply:
Actually, the right way probably involves multi-nesting

```
if (!punishment) {
  room = ...
  if (room.parent) {
```
etc